bootstrap/updater.go

```diff
@@ -0,0 +1,406 @@
+package bootstrap
+
+import (
+	"context"
+	"crypto/ed25519"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/santhosh-tekuri/jsonschema/v5"
+	"github.com/spf13/afero"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log"
+)
+
+const (
+	DefaultURI = "http://localhost:3000/bootstrap"
+	DirName    = "bootstrap"
+
+	timeout    = 5 * time.Second
+	schemaFile = "schema.json"
+	format     = "2006-01-02T15-04-05"
+)
+
+var (
+	ErrEpochOutOfOrder  = errors.New("epoch out of order")
+	ErrWrongVersion     = errors.New("wrong schema version")
+	ErrInvalidSignature = errors.New("invalid signature")
+	ErrInvalidBeacon    = errors.New("invalid beacon")
+)
+
+type realClient struct{}
+
+func (realClient) Query(ctx context.Context, resource *url.URL) ([]byte, error) {
+	req, err := http.NewRequestWithContext(ctx, "GET", resource.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("create http request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	resp, err := (&http.Client{}).Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("http get bootstrap file: %w", err)
+	}
+	defer resp.Body.Close()
+	data, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("bootstrap read resonse: %w", err)
+	}
+	return data, nil
+}
+
+type Config struct {
+	URI             string `mapstructure:"bootstrap-uri"`
+	Version         string `mapstructure:"bootstrap-version"`
+	SpacemeshPubKey string `mapstructure:"spacemesh-pubkey"`
```

Comment:
i think the agreement is that we won't be doing it like this. signature comes from PKI and HTTPS, so it will be verified when TLS connection is establishedd

Reply:
you are right. it was a previous version where i wanted to merge recovery use case with bootstrapping. 
removed.